web/js/components/timeline/timeline-controls/date-change-arrows.js

```diff
@@ -1,82 +1,95 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import HoverTooltip from '../../util/hover-tooltip';
+import PreloadSpinner from '../../animation-widget/preload-spinner';
+import {
+  setArrowDown as setArrowDownAction,
+  setArrowUp as setArrowUpAction,
+} from '../../../modules/date/actions';
 
-const ANIMATION_DELAY = 200; // interval firing to trigger parent level arrow change
-const CLICK_TIMEOUT_DELAY = 500; // wait before click becomes a delay
+const ANIMATION_DELAY = 600; // interval for timestep
+const CLICK_HOLD_DELAY = 200; // wait before click is considered a hold
 
-let mouseHoldCheckTimer = null;
-let isMouseHolding = false;
+let arrowDownCheckTimer = null;
 // left/right arrow intervals
 const intervals = {
   left: 0,
   right: 0,
 };
 
 class DateChangeArrows extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.arrowDownMap = {
+      left: props.leftArrowDown,
+      right: props.rightArrowDown,
+    };
+  }
+
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleKeyDown);
+    document.addEventListener('keyup', this.handleKeyUp);
+  }
+
+  componentDidUpdate (prevProps) {
+    const { tilesPreloaded, arrowDown } = this.props;
```

Comment:
May want to clarify as `arrowDownDirection`.

Reply:
Yeah it's kind of an overloaded variable but that was intentional.  If it evaluates as `''` then no arrow is being held down, otherwise it's `left` or `right` for which direction is being held down.  I think this is more concise than having a `isArrowDown` boolean and an `arrowDownDirection` for example.